server/utils/nunjucksSetup.ts

```diff
@@ -245,5 +253,35 @@ export function registerNunjucks(conditionService: ConditionService, app?: expre
     }
   })
 
+  njkEnv.addFilter('dateToDisplay', (licence: Licence) => {
+    const licenceType = licence.typeCode
+    const led = licence.licenceExpiryDate ? toDate(licence.licenceExpiryDate) : null
+    const tussd = licence.topupSupervisionStartDate ? toDate(licence.topupSupervisionStartDate) : null
+
+    let dateToDisplay: Date
+    let textToDisplay = ''
+
+    if (licenceType === 'AP') {
+      textToDisplay = 'Licence end date'
+      dateToDisplay = led
+    }
+
+    if (licenceType === 'AP_PSS') {
+      textToDisplay = isToday(tussd) ? 'PSS end date' : 'Licence end date'
+      dateToDisplay = led
```

Comment:
Should this not be `isToday(tussd) ? tussd: led`?

Reply:
that's what I thought at first but when I re-read it. i thought it meant the title should change, not the actual date

Reply:
I think that's something we'd need to check, otherwise surely we would be showing the wrong date with the wrong label? Eg LED with the label of PSS End Date?

Reply:
changes made following clarification from BA.